### ISSUE TITLE
Parallel consistency

### ISSUE THREAD

**Issue by janmandel** [openwfm/WRF-SFIRE#13]
Parallel testing shows small differences with with diferent processor grids on testing_GFSF with fire on. But all is good when the fire is off, cf. #12

**Comment by janmandel** [comment]
/glade/u/home/jmandel/p/testing-GFSF/wrf-10x10
/glade/u/home/jmandel/p/testing-GFSF/wrf-8x12
/glade/u/home/jmandel/p/testing-GFSF/wrf-9x9
using /glade/u/home/jmandel/work/WRF-SFIRE-v42/main/wrf.exe at 1577f6c034c91bf7a8372a2ae8a96e79bad36666

**Comment by janmandel** [comment]
psfc_old in frame 2 has a line of zeros for 9x9 processor mesh but not 10x10 mesh!
<img src="https://user-images.githubusercontent.com/6204560/85191305-96dcdd80-b27b-11ea-8cae-2b6971718fa9.png" width="200" height="200"/> <img src="https://user-images.githubusercontent.com/6204560/85191314-9f351880-b27b-11ea-8464-8a6f0fa657a3.png" width="200" height="200"/> <img src="https://user-images.githubusercontent.com/6204560/85191318-a2c89f80-b27b-11ea-9160-199be5d3f48f.png" width="200" height="200"/>

**Comment by janmandel** [comment]
Building with debug prints for psfc_old in /glade/work/jmandel/WRF-SFIRE-mpi at a94aa33a3cb8f2c1ca9c2a06613794dd78a59023

**Comment by janmandel** [comment]
The values of PSFC_OLD seem to get corrupt in call to advance_moisture:
```
PSFC     in  11  1 0.86E+05  2 0.86E+05  3 0.86E+05  4 0.86E+05  5 0.86E+05  6 0.86E+05  7 0.86E+05  8 0.86E+05  9 0.86E+05 10 0.86E+05 11 0.86E+05
PSFC_OLD in  11  1 0.00E+00  2 0.00E+00  3 0.00E+00  4 0.00E+00  5 0.00E+00  6 0.00E+00  7 0.00E+00  8 0.00E+00  9 0.00E+00 10 0.00E+00 11 0.00E+00
SFIRE:advance_moisture start
SFIRE:advance_moisture dt=      0.00s using    5 classes from possible    5
SFIRE:T2                         1  11   1  11   282.       283.       283.
SFIRE:Q2                         1  11   1  11  0.431E-02  0.488E-02  0.468E-02
SFIRE:PSFC                       1  11   1  11  0.856E+05  0.858E+05  0.857E+05
SFIRE:advance_moisture: initializing, copying surface variables to old
SFIRE:minimal    t2  0.282E+03  q2  0.431E-02  psfc  0.856E+05
SFIRE:floor      t2  0.200E+03  q2  0.100E-07  psfc  0.100E+04
SFIRE:Rain intensity    min      0.00 max      0.00 mm/h
SFIRE:Relative humidity min     23.96 max     57.18%
SFIRE: 1 equilibrium fmc_equi    1  11   1  11  0.929E-01  0.160      0.149
SFIRE: 2 equilibrium fmc_equi    1  11   1  11  0.929E-01  0.160      0.149
SFIRE: 3 equilibrium fmc_equi    1  11   1  11  0.929E-01  0.160      0.149
SFIRE: 4 equilibrium fmc_equi    1  11   1  11  0.929E-01  0.160      0.149
SFIRE: 5 equilibrium fmc_equi    1  11   1  11  0.100      0.168      0.158
SFIRE: 1 time lag                1  11   1  11   1.00       1.00       1.00
SFIRE: 2 time lag                1  11   1  11   10.0       10.0       10.0
SFIRE: 3 time lag                1  11   1  11   100.       100.       100.
SFIRE: 4 time lag                1  11   1  11  0.100E+04  0.100E+04  0.100E+04
SFIRE: 5 time lag                1  11   1  11  0.100E+10  0.100E+10  0.100E+10
SFIRE: 1 after advance fmc_gc    1  11   1  11  0.929E-01  0.160      0.149
SFIRE: 2 after advance fmc_gc    1  11   1  11  0.929E-01  0.160      0.149
SFIRE: 3 after advance fmc_gc    1  11   1  11  0.929E-01  0.160      0.149
SFIRE: 4 after advance fmc_gc    1  11   1  11  0.929E-01  0.160      0.149
SFIRE: 5 after advance fmc_gc    1  11   1  11  0.300      0.300      0.300
SFIRE:advance_moisture end
PSFC_OLD out 11  1 0.49E-02  2 0.49E-02  3 0.49E-02  4 0.49E-02  5 0.48E-02  6 0.48E-02  7 0.48E-02  8 0.48E-02  9 0.48E-02 10 0.48E-02 11 0.86E+05
```

**Comment by janmandel** [comment]
??
```
 copy2old start
PSFC     in  11  1 0.86E+05  2 0.86E+05  3 0.86E+05  4 0.86E+05  5 0.86E+05  6 0.86E+05  7 0.86E+05  8 0.86E+05  9 0.86E+05 10 0.86E+05 11 0.86E+05
PSFC_OLD in  11  1 0.00E+00  2 0.00E+00  3 0.00E+00  4 0.00E+00  5 0.00E+00  6 0.00E+00  7 0.00E+00  8 0.00E+00  9 0.00E+00 10 0.00E+00 11 0.00E+00
PSFC_OLD out 11  1 0.49E-02  2 0.49E-02  3 0.49E-02  4 0.49E-02  5 0.48E-02  6 0.48E-02  7 0.48E-02  8 0.48E-02  9 0.48E-02 10 0.48E-02 11 0.86E+05
 copy2old end
```

**Comment by janmandel** [comment]
???
```
 copy2old start
PSFC     in  11  1 0.86E+05  2 0.86E+05  3 0.86E+05  4 0.86E+05  5 0.86E+05  6 0.86E+05  7 0.86E+05  8 0.86E+05  9 0.86E+05 10 0.86E+05 11 0.86E+05
PSFC_OLD in  11  1 0.00E+00  2 0.00E+00  3 0.00E+00  4 0.00E+00  5 0.00E+00  6 0.00E+00  7 0.00E+00  8 0.00E+00  9 0.00E+00 10 0.00E+00 11 0.00E+00
PSFC         11  1 0.86E+05
PSFC_OLD set 11  1 0.86E+05
PSFC         11  2 0.86E+05
PSFC_OLD set 11  2 0.86E+05
PSFC         11  3 0.86E+05
PSFC_OLD set 11  3 0.86E+05
PSFC         11  4 0.86E+05
PSFC_OLD set 11  4 0.86E+05
PSFC         11  5 0.86E+05
PSFC_OLD set 11  5 0.86E+05
PSFC         11  6 0.86E+05
PSFC_OLD set 11  6 0.86E+05
PSFC         11  7 0.86E+05
PSFC_OLD set 11  7 0.86E+05
PSFC         11  8 0.86E+05
PSFC_OLD set 11  8 0.86E+05
PSFC         11  9 0.86E+05
PSFC_OLD set 11  9 0.86E+05
PSFC         11 10 0.86E+05
PSFC_OLD set 11 10 0.86E+05
PSFC         11 11 0.86E+05
PSFC_OLD set 11 11 0.86E+05
PSFC_OLD out 11  1 0.49E-02  2 0.49E-02  3 0.49E-02  4 0.49E-02  5 0.48E-02  6 0.48E-02  7 0.48E-02  8 0.48E-02  9 0.48E-02 10 0.48E-02 11 0.86E+05
 copy2old end
```

**Comment by janmandel** [comment]
/glade/scratch/jmandel/testing/wrf-10x10.2 with file_print_msg=1 still getting stuck on computing sum of all tracers #9

**Comment by janmandel** [comment]
1470b176036d293a59a298d658b018b4a775016d PSFC_OLD now constent, but
```
[r,s,n]=ncdiff('wrf-10x10.2/wrfout_d03_2020-05-20_15:00:00','wrf-9x9.2/wrfout_d03_2020-05-20_15:00:00')
>> e={n{find(p)}}
e =
  1x13 cell array
  Columns 1 through 8
    {'F_LINEINT'}    {'F_LINEINT2'}    {'F_ROS'}    {'F_ROS13'}    {'F_ROS21'}    {'F_ROS23'}    {'F_ROS31'}    {'F_ROS32'}
  Columns 9 through 13
    {'F_ROS33'}    {'F_ROSX'}    {'F_ROSY'}    {'UF'}    {'VF'}
```

**Comment by janmandel** [comment]
Increasing halos did not help. But these runs were with FIRE_WIND_LOG_INTERP=4 which calls interpolate_atm2fire, which uses uah and vah, which were not allocated, another memory error because of registry packaging - uah vah were in package ifire=2 but not ifire=1.
FIRE_WIND_LOG_INTERP not 4, which calls  interpolate_wind2fire_height and does not use uah and  vah, should not be affected but it needs to be tested.

**Comment by janmandel** [comment]
![uf10-9spy](https://user-images.githubusercontent.com/6204560/85213578-74030580-b31d-11ea-9217-e2a37546a046.png)
![vf10-9spy](https://user-images.githubusercontent.com/6204560/85213567-50d85600-b31d-11ea-8df9-af1124aa227a.png)
![uah10-9spy](https://user-images.githubusercontent.com/6204560/85213571-59309100-b31d-11ea-8736-b74ca9806448.png)
![vah10-9spy](https://user-images.githubusercontent.com/6204560/85213575-5fbf0880-b31d-11ea-9a7b-22ea29d23b25.png)

**Comment by janmandel** [comment]
All is good with fire_wind_log_interp = 1, which was fixed in #10.
The previous default fire_wind_log_interp = 4 (with wind reduction factors) and motivated #13 is obsolete and not parallel consistent. Is it worth fixing? #23
Changed default fire_wind_log_interp = 1 in eb7773f6088b1cddc36ae640b06c400b1b2d8a94
Change the default also in openwfm/wrfxpy#52
